src/domain/interface.ts

```diff
@@ -178,8 +178,8 @@ export interface ApplicationSummary {
   isAttestable: boolean;
   ableToRenew: boolean;
   lastPausedAtUtc?: Date;
-  sourceAppId?: string;
-  renewalAppId?: string;
+  sourceAppId?: string | null;
+  renewalAppId?: string | null;
```

Comment:
> is null being considered to accommodate an external lib (mongoose maybe)?

I checked this out, when an `undefined` value is passed to a mongoose update call it will coerce it to `null` - I saw this behaviour with the unsetting of `pauseReason` but hadn't looked it up in the docs. This behaviour changes in mongoose 6 (dac-api is on 5.12.6 rn), `undefined` values will get removed from update calls altogether: https://mongoosejs.com/docs/migrating_to_6.html#removed-omitundefined
So looks like it may be a good idea to change these `field = undefined` to `field = null` instead, if I want to have nulls. In the case of `pauseReason` it might be fine to unset the field, for these renewal + source appIds i'm wondering if it's useful to have them as null (so it's known they've existed at some point).

Reply:
> This isn't part of your changes, but why is `Application` and `ApplicationSummary` included as separate interfaces? Could `Application` extend `ApplicationSummary`?

I think they're different enough to warrant separate interfaces - i.e. there are some fields in the summary that are parsed versions of parts of application Sections, so the nesting is different, or even the the type itself (like `currentApprovedAppDoc` is a boolean). But maybe it would useful to have a base type both of them could extend from? For fields like `appId`, `submitterId`, date fields, etc...

Reply:
I added an `ApplicationBaseFields` interface that `Application` and `ApplicationSummary` can extend